go/database/mpt/live_trie.go

```diff
@@ -63,18 +63,38 @@ func OpenFileLiveTrie(directory string, config MptConfig, cacheCapacity int) (*L
 // VerifyFileLiveTrie validates a file-based live trie stored in the given
 // directory. If the test passes, the data stored in the respective directory
 // can be considered to be a valid Live Trie of the given configuration.
-func VerifyFileLiveTrie(directory string, config MptConfig, observer VerificationObserver) error {
+func VerifyFileLiveTrie(directory string, config MptConfig, observer VerificationObserver) (res error) {
+	if observer == nil {
+		observer = NilVerificationObserver{}
+	}
+
 	metadata, exists, err := readMetadata(directory + "/meta.json")
 	if err != nil {
 		return err
 	}
 	if !exists {
 		return nil
 	}
-	return VerifyFileForest(directory, config, []Root{{
+	observer.StartVerification()
+	defer func() {
+		if r := recover(); r != nil {
+			panic(r)
+		}
+		observer.EndVerification(res)
+	}()
+
+	// Open stock data structures for content verification.
+	observer.Progress("Obtaining read access to files ...")
+	source, err := openVerificationNodeSource(directory, config)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
```

Comment:
This code is duplicated in the Archive and LiveTrie implementation.

I would suggest to rename the current `verifyFileForest` accepting a `*verificationNodeSource` to `verifyForest` and create a new `verifyFileForest` function that handles the source opening and closing and calls the `verifyForest` function for the actual checks.

Reply:
This was caused by calling `verifyFileForest` instead of `VerifyMptState`. This duplicity got removed by calling the correct func.